### PR TITLE
Clean nit compilation directory

### LINKED ISSUE #792: Default nit compilation directory
The question that follow #789 is the default location of the .nit_compile directory.
The current working directory may not be the best place.

Some sensible choices are:

* in tmp (like gcc and other tools). But unlike gcc the name of the file must be stable to help ccache. However, /tmp is usually a ramdisk so better performances is a possible benefit.
* in ~ (or ~/.cache). So the user share its own stuff.
* in the root of the project (may not work if the owner is not the user)
* in the directory of the module (same issue)
* in the current working directory (status quo)
* in the current working directory but not hidden, and we name it `NIT_COMPILE_DIRECTORY_BEUAAARG` because we are proud

Other question, should nitg clean the directory after it? (with an optional --no-clean option for debugging the generated code)

### PR DESCRIPTION
**Pull request by privat**
This PR cause the compilation directory to be removed after compilation.

This auto-removal is disabled if `--compile-dir` or `--no-cc` is used because it is a sane default I think.
Moreover, the compilation directory is renamed by defaut to `nit_compile` because there is no more point to make it hidden since its presence means an active request (`--no-cc`).

The PR is a bit premature since the boostrap will fail since `rmdir` requires the light FFI, so it is planned to be merged after the regen of c_src.

Close #792

### CODE REVIEW

**Comment by Morriar** [comment]
No `gitignore` update?

**Comment by xymus** [comment]
I like the idea to remove the folder, but not the name change. Even if normal devs are never supposed to see it, compiler devs like us will see and use it. I need to keep it around until I test a mobile app for example. Removing the dot is ok, but please, no all caps... no need to screen at every `ls`.

Even some end-users (devs) will need it when writing Android applications. The Java language does not support redirecting error to their original source in Nit FFI code, so they need to refer to the generated/extracted code in `NIT_COMPILE/android/src/`. I can't see myself adding this to the documentation.

Plus, telling us the path to the generated files is `NIT_COMPILE` implies that it is an env var.

**Comment by privat** [comment]
@xymus I agree. I will dowcase the name

**Comment by privat** [comment]
reroll: use lowcased `nit_compile` as the default compilation directory.

**Comment by BlackMinou** [comment]
+1

**Comment by privat** [comment]
update: remove reference to .nit_compile in other files, Makefile and .git_ignore; explicitly clean in tests.sh (because --compile-dir is used)

**Comment by xymus** [comment]
:shipit:

**Comment by privat** [comment]
reroll&update: cleaning of the toolchain-compiler relation and distinguish `compile_dir` where C files are stored and `root_compile_dir` that might be a parent directory on some plateforms.

**Comment by xymus** [comment]
Still +1

File: share/man/nitc.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ By default, the generated executables are produced in the current directory.
 
 Internally, nitc rely on the presence of a C compiler. Usually gcc (but nitc was successfully tested with clang).
 A compilation directory is therefore created and (re-)used.
-By default, the compilation directory is named `.nit_compile`.
+By default, the compilation directory is named `nit_compile` and in removed after the compilation.
```
Comment (R4PaSs) [inline share/man/nitc.md]: :s/in/is

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
reroll: spell in manpages

**Pull request merged**